component/org.wso2.carbon.identity.api.otp.service.emailotp/src/main/java/org/wso2/carbon/identity/api/otp/service/emailotp/impl/EmailotpApiServiceImpl.java

```diff
@@ -0,0 +1,110 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.api.otp.service.emailotp.impl;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.extension.identity.emailotp.common.dto.FailureReasonDTO;
+import org.wso2.carbon.extension.identity.emailotp.common.dto.GenerationResponseDTO;
+import org.wso2.carbon.extension.identity.emailotp.common.dto.ValidationResponseDTO;
+import org.wso2.carbon.extension.identity.emailotp.common.exception.EmailOtpClientException;
+import org.wso2.carbon.extension.identity.emailotp.common.exception.EmailOtpException;
+import org.wso2.carbon.identity.api.otp.service.emailotp.EmailotpApiService;
+import org.wso2.carbon.identity.api.otp.service.emailotp.dto.OTPGenerateResponse;
+import org.wso2.carbon.identity.api.otp.service.emailotp.dto.OTPGenerationRequest;
+import org.wso2.carbon.identity.api.otp.service.emailotp.dto.OTPValidationFailureReason;
+import org.wso2.carbon.identity.api.otp.service.emailotp.dto.OTPValidationRequest;
+import org.wso2.carbon.identity.api.otp.service.emailotp.dto.OTPValidationResponse;
+import org.wso2.carbon.identity.api.otp.service.emailotp.util.EndpointUtils;
+
+import javax.ws.rs.core.Response;
+
+/**
+ * This class implements the service layer for
+ * org.wso2.carbon.identity.api.otp.service.emailotp.EmailotpApi.
+ */
+public class EmailotpApiServiceImpl implements EmailotpApiService {
+
+    private static final Log log = LogFactory.getLog(EmailotpApiServiceImpl.class);
+
+    /**
+     * This method is implemented from org.wso2.carbon.identity.api.otp.service.emailotp.EmailotpApi
+     * to generate OTP.
+     *
+     * @param otpGenerationRequest Otp generation request object.
+     * @return Response
+     */
+    @Override
+    public Response emailotpGeneratePost(OTPGenerationRequest otpGenerationRequest) {
+
+        String userId = StringUtils.trim(otpGenerationRequest.getUserId());
+        try {
+            GenerationResponseDTO responseDTO = EndpointUtils.getEmailOTPService().generateEmailOTP(userId);
+            OTPGenerateResponse response = new OTPGenerateResponse()
+                    .transactionId(responseDTO.getTransactionId())
+                    .emailOtp(responseDTO.getEmailOTP());
+            return Response.ok(response).build();
+        } catch (EmailOtpClientException e) {
+            return EndpointUtils.handleBadRequestResponse(e, log);
+        } catch (EmailOtpException e) {
+            return EndpointUtils.handleServerErrorResponse(e, log);
+        } catch (Throwable e) {
+            return EndpointUtils.handleUnexpectedServerError(e, log);
+        }
+    }
+
+    /**
+     * This method is implemented from org.wso2.carbon.identity.api.otp.service.emailotp.EmailotpApi
+     * to validate OTP.
```

Comment:
Let's follow javadoc guidelines for comments. 
For an example, we can use ` {@link EmailotpApi} ` here

Reply:
Removed those because there was an issue with those in PR-builder.